### DISCUSSION TITLE
OR condition not working for certain cases

### DISCUSSION THREAD

**Discussion by kesavamuthu** [prisma/prisma#23623] [Q&A] [answered]
### Used OR condition but query which generated it's showing AND condition instead of OR. It's not working for this model alone I think because I've used OR conditions in some other models also that's working.
```js
prisma.comments.findMany({
    where: {
      OR: [{ uuid: uuid, status: 1 }],
    },
  })
```
Generated query:

SELECT `db_listicle_v1`.`comments`.`id`, `db_listicle_v1`.`comments`.`content`, `db_listicle_v1`.`comments`.`updatedBy`, `db_listicle_v1`.`comments`.`createdBy`, `db_listicle_v1`.`comments`.`updatedOn`, `db_listicle_v1`.`comments`.`deletedOn`, `db_listicle_v1`.`comments`.`createdOn`, `db_listicle_v1`.`comments`.`parent`, `db_listicle_v1`.`comments`.`slug`, `db_listicle_v1`.`comments`.`read`, `db_listicle_v1`.`comments`.`mentions`, `db_listicle_v1`.`comments`.`uuid`, `db_listicle_v1`.`comments`.`uuidTemp`, `db_listicle_v1`.`comments`.`url`, `db_listicle_v1`.`comments`.`jduid`, `db_listicle_v1`.`comments`.`source`, `db_listicle_v1`.`comments`.`status`, `db_listicle_v1`.`comments`.`moderatedBy`, `db_listicle_v1`.`comments`.`moderatedAt` FROM `db_listicle_v1`.`comments` WHERE (`db_listicle_v1`.`comments`.`uuid` = ? AND `db_listicle_v1`.`comments`.`status` = ?)

### Expected behavior (optional)

simple OR condition should be applied.

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
model comments {
  id               Int           @id @default(autoincrement())
  content          String?       @db.Text
  updatedBy        Int?
  createdBy        Int?
  updatedOn        DateTime?     @default(dbgenerated("(utc_timestamp())")) @db.Timestamp(0)
  deletedOn        DateTime?     @db.Timestamp(0)
  createdOn        DateTime?     @default(dbgenerated("(utc_timestamp())")) @db.Timestamp(0)
  parent           Int?
  slug             String        @db.VarChar(250)
  read             Int?          @default(1) @db.TinyInt
  mentions         Json?
  uuid             Bytes?        @db.Binary(16)
  uuidTemp         Bytes?        @db.Binary(16)
  url              String?       @db.VarChar(450)
  jduid            String?       @db.VarChar(30)
  source           Int?          @default(0) @db.TinyInt
  status           Int?          @default(0) @db.TinyInt
  moderatedBy      Int?
  moderatedAt      DateTime?     @db.Timestamp(0)
  createdByTousers users?        @relation("comments_createdByTousers", fields: [createdBy], references: [id], onDelete: NoAction, onUpdate: NoAction, map: "fk_comments_createdBy")
  updatedByTousers users?        @relation("comments_updatedByTousers", fields: [updatedBy], references: [id], onDelete: NoAction, onUpdate: NoAction, map: "fk_comments_updatedBy")
  articles         articles?     @relation(fields: [uuid], references: [uuid], onUpdate: NoAction, map: "fk_comments_uuid")
  articlesTemp     articlesTemp? @relation(fields: [uuidTemp], references: [uuid], onUpdate: NoAction, map: "fk_comments_uuid_temp")
  parentComment    comments?     @relation("commentsTocomments", fields: [parent], references: [id], onDelete: NoAction, onUpdate: NoAction, map: "fk_parent_comments")
  replies          comments[]    @relation("commentsTocomments")

  @@index([createdBy], map: "fk_comments_createdBy_idx")
  @@index([updatedBy], map: "fk_comments_updatedBy_idx")
  @@index([parent], map: "fk_root_comments_idx")
  @@index([uuid], map: "fk_comments_uuid")
  @@index([uuidTemp], map: "fk_comments_uuid_temp")
}

```

```
- OS: Linux  3.10.0-1160.el7.x86_64 #1 SMP Mon Oct 19 16:18:59 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux

- Database:MySQL
- Node.js version: v16.14.1

- prisma                  : 4.12.0
@prisma/client          : 4.12.0
Current platform        : rhel-openssl-1.0.x
Query Engine (Node-API) : libquery-engine 659ef412370fa3b41cd7bf6e94587c1dfb7f67e7 (at node_modules/@prisma/engines/libquery_engine-rhel-openssl-1.0.x.so.node)
Migration Engine        : migration-engine-cli 659ef412370fa3b41cd7bf6e94587c1dfb7f67e7 (at node_modules/@prisma/engines/migration-engine-rhel-openssl-1.0.x)
Format Wasm             : @prisma/prisma-fmt-wasm 4.12.0-67.659ef412370fa3b41cd7bf6e94587c1dfb7f67e7
Default Engines Hash    : 659ef412370fa3b41cd7bf6e94587c1dfb7f67e7
Studio                  : 0.483.0
Preview Features        : orderByNulls, filteredRelationCount, extendedWhereUnique

```

**Comment by ludralph** [comment, marked as answer]
Hi @kesavamuthu 👋 

Thank you for raising this question.

The issue you're facing seems to be related to the way you're using the `OR` condition in your Prisma query. In Prisma, the OR condition expects an array of objects, where each object represents a condition. However, in your query, you're passing a single object to the OR condition. Restructure your query as shown below:
```js
prisma.comments.findMany({
    where: {
      OR: [
        { uuid: uuid },
        { status: 1 }
      ],
    },
})
```
In this query, the `OR` condition is an array of two objects. The first object has the condition `uuid: uuid`, and the second object has the condition `status: 1`. This query will return all comments where the uuid is equal to the provided uuid or the status is 1. This was also discussed in this closed [issue](https://github.com/prisma/prisma/issues/18269).

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by kesavamuthu** [reply]
Hi, @ludralph felt like from typescript end it needs to show some error for the incorrect syntax.
